File: command/command/document/paragraph/ParagraphMemento.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace command.document
{
	public class ParagraphMemento : IMemento
	{
		private readonly IParagraph _paragraph;
		private readonly string _text;

		public ParagraphMemento(IParagraph paragraph, string text)
		{
			_paragraph = paragraph ?? throw new ArgumentNullException(nameof(paragraph));
			_text = text ?? throw new ArgumentNullException(nameof(text));
		}

		public void Restore() => _paragraph.Text = _text;
	}
}

File: command_java/Document/src/document/documentItem/paragraph/ParagraphMemento.java
package document.documentItem.paragraph;

import hsitory.IMemento;

public class ParagraphMemento implements IMemento {

    public ParagraphMemento(String oldText, IMutableParagraph paragraph) {
        _paragraph = paragraph;
        _oldText = oldText;
    }

    @Override
    public void Restore() {
        _paragraph.SetText(_oldText);
    }

    private final IMutableParagraph _paragraph;
    private final String _oldText;
}
